crates/controller/src/hud_interaction.rs

```diff
@@ -0,0 +1,32 @@
+use bevy::prelude::*;
+
+use de_core::objects::BuildingType;
+use de_spawner::ObjectCounter;
+use crate::{
+    command::place_draft_system,
+    draft::NewDraftEvent,
+    pointer::Pointer,
+};
+
+#[derive(Component, Clone, Copy)]
+pub(crate) enum HudButtonAction {
+    Build(BuildingType),
+}
+
+pub(crate) fn hud_button_system(
+    interactions: Query<(&Interaction, &HudButtonAction), Changed<Interaction>>,
+    counter: Res<ObjectCounter>,
+    pointer: Res<Pointer>,
```

Comment:
It is clear that the pointer is over HUD -> I would change initial placement to the middle of the screen. Note that this might require modification of the camera crate to somehow expose focus terrain point (it is already there but not public and with nice API).